app/scripts/metamask-controller.js

```diff
@@ -1700,7 +1698,6 @@ export default class MetamaskController extends EventEmitter {
         this.preferencesController.store.getState()
           ?.disabledRpcMethodPreferences?.eth_sign,
       getAllState: this.getState.bind(this),
-      securityProviderRequest: this.securityProviderRequest.bind(this),
```

Comment:
Can you open up a task to clean up the SignatureController in core?

Reply:
https://github.com/MetaMask/core/issues/4102